packages/chat-stateful-client/src/ChatClientState.ts

```diff
@@ -14,6 +14,12 @@ export type ChatClientState = {
    * Object with {@Link ChatThreadClientState} fields, keyed by {@Link ChatThreadClientState.threadId}.
    */
   threads: { [key: string]: ChatThreadClientState };
+  /**
+   * Stores the latest error for each API method.
+   *
+   * See documentation of {@Link ChatErrors} for details.
+   */
+  latestErrors: ChatErrors;
```

Comment:
As @JamesBurnside had recommended, limiting both the stateful and adapter state to store only the latest errors.

In contrast, the on('error') event from adapter will trigger for each error.

There's no demonstrated use-case for storing more than one error per method and it adds a lot of complexity both for Contoso and the implementation to deal with lists of errors (e.g., when do we clear the list / who does).